Fix scale_factor_override with cursor position.

ahmedcharles opened the pull request:
Sometimes the wrong scale factor was used (not accounting for
overrides).

Fixes #2501

cart commented:
Nice catch! This implementation works, but to enable more flexibility and increased consistency, I think we should probably adopt our Window type's approach to width / physical_width here:

1. Only store the physical_cursor_position on Window
2. Add a physical_cursor_position getter method on Window.
3. Make `Window::cursor_position()` return cursor_position / scale_factor (aka the "logical" position). Just look at `Window::width()` and mirror that. 

Then when handling the winit events, all we need to do is:

1. Set the "physical cursor position" when it changes
2. Read the `Window::cursor_position()` from the window (to reuse the logic there).

swwind commented:
Why this PR was closed? 🤔

I think it works fine for me.

ahmedcharles commented:
Note: the cart said that it works above.

I closed it because I don't plan on addressing the comments, as is my prerogative. This PR could've had two responses:

1. Thanks, this works, I'll merge it. If you're looking for something else to do, please consider X.
2. Thanks, this works, but I'll hold your contribution hostage until you resolve X.

Whether you think my characterization is fair or not, in specifics, the first one above clearly has no downside while showing new contributors that their time is valuable and the project avoids scope creep, at least for simple things. I don't plan on contributing to projects that don't value my time.
